File: 리트코드/TwoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        num2 = nums[:]; 
        answer = [] ; 
        
        for index,value in enumerate(num2):
            num2[index] = target - num2[index] ; 
        
        
        for index2, value2 in enumerate(num2):
            for index,value in enumerate(nums):
                if index2 == index : 
                    continue; 
                if num2[index2] == nums[index] :
                    answer.append(index2); 
                    answer.append(index); 
                    return answer; 
            
            
        

File: 리트코드/TwoSum.cpp
// https://leetcode.com/problems/two-sum/submissions/
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        vector<int> num2(nums.size()) ; 
        vector<int> ans ; 
        for(int i = 0 ; i< nums.size() ; i++){
            num2[i] = target -nums[i]  ; 
            //cout<<num2[i] <<endl; 
        }
        
        for(int i =0 ; i< num2.size() ; i++){
            for(int j = 0 ; j< nums.size() ; j++){
                if(i == j) continue;  // [3,2,4] , 6 일 때 [0,0]을 답으로 멍청하게 내놓지 않게끔.. 
                if(num2[i] == nums[j]){
                    //cout<< num2[i] << "," <<nums[j] <<endl; 
                    ans.push_back(i); 
                    ans.push_back(j); 
                    
                    return ans; 
                }
            }
        }
        return ans; 
    }
};
